docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,606 @@
+`npe2` as soon as possible.  See the [](npe2-migration-guide).
+```
+
+The **plugin manifest** is a specially formatted text file declaring the
+functionality of a [npe2][] plugin. A **plugin** is a python package that
+contains the manifest together with a suitable _[entry point group][epg]_ in
+the package metadata.
+
+Manifest files may be [json][], [yaml][], or [toml][] files conforming to the
+manifest schema. The **schema** defines what to expect in a manifest by
+defining the fields and their data types. These fields and their meanings are
+described below.
+
+A **plugin engine** is used to discover plugins, provide utilities for
+querying and manipulating plugins, and for exposing plugin-backed
+functionality to _napari_. **Discovery** is the process that finds plugins,
+parses the manifests and indexes them for later use. The [npe2][] library
+manages these responsibilities.
+
+## Configuring a python package to use a plugin manifest
+
+### 1. Add package metadata for locating the manifest
+
+The manifest file should be specified in the plugin's `setup.cfg` or
+`setup.py` file using the _[entry point group][epg]_: `napari.manifest`. For
+example, this would be the section for a plugin `npe2-tester` with
+`napari.yaml` as the manifest file:
+
+```cfg
+[options.entry_points]
+napari.manifest =
+    npe2-tester = npe2_tester:napari.yaml
+```
+
+The manifest file is specified relative to the submodule root path.
+So for the example it will be loaded from: `<path/to/npe2-tester>/napari.yaml`.
+
+### 2. Include the manifest in the package distribution
+
+The manifest file needs to be included as _[package data][pd]_ in
+distributable forms for the package. For example:
+
+```toml
+[metadata]
+...
+include_package_data=True
+
+[options.package_data]
+npe2_tester =
+    napari.yaml
+```
+
+## Manifest schema
+
+The plugin manifest file is a [json][], [yaml][], or [toml][] conforming to
+a schema. The `npe2` package includes a command-line tool that can be used
+to validate a schema:
+
+```
+pip install npe2
+npe2 validate my_plugin_manifest.yml
+```
+
+The manifest file's fields are described in detail below. The manifest file's
+overall structure can be viewed as a hierarchy:
+
+```
+<top-level properties>
+contributions:
+  commands:     <list of commands>
+  readers:      <list of readers>
+  writers:      <list of writers>
+  sample_data:  <list of sample data providers>
+  widgets:      <list of widget providers>
+  themes:       <list of themes>
+```
+
+Readers, writers, sample data providers and widget providers refer to callable
+python functions that a plugin defines. Each callable is identified with an
+entry in the list of `commands` via a unique id. For more see {ref}`Commands`
+below.
+
+```{note}
+Python package metadata (`setup.py` or `setup.cfg`) may be used to populate
+some internal properties of the plugin manifest. These include the authors,
+description, and version.
+```
+
+(top-level-props)=
+
+## Top-level properties
```

Comment:
I think `Contributions` should have its own page, rather than each individual subsection of contributions appearing as a flat header next to "top level props"

Reply:
I think this page is extremely easy to navigate: https://code.visualstudio.com/api/references/contribution-points